### ISSUE TITLE
Does VirtualBox recognize that virtual camera?

### ISSUE THREAD

**Issue by MakarkinPRO** [CatxFish/obs-virtual-cam#11]
Does VirtualBox recognize that virtual camera?

**Comment by CatxFish** [comment]
No, it can't. Only an usb device can be recognized by VirtualBox.

**Comment by MakarkinPRO** [comment]
Any idea of how to connect that stuff? My main purpose is to stream my presentation from pc to Instagram.

**Comment by CatxFish** [comment]
It seems like Genymotion can detect camera  , but I can't active successfully now.
I will try to test it if I am free.

**Comment by MakarkinPRO** [comment]
I have it too. But it see only webcam0 and webcam1 which is OBS virtual?

**Comment by MakarkinPRO** [comment]
Heh tried 2 days. Still no results. Somebody tell what, that virtual camera should me kind of USB camera to VirtualBox and then Android Emulator may see it.

**Comment by CatxFish** [comment]
Still can't find the reason why Genymotion do not work fine. 
I will give a update if I find a workable solution.
Apply this plugin to an android app is a cool idea,  I really want it success.

**Comment by MakarkinPRO** [comment]
Hm. Still have a https://www.screencast.com/t/zotV6X6Femy grenn screen

**Comment by CatxFish** [comment]
I haven't make a full test on Genymotion compatibility . So I can't tell you what is the actually rule .
But here is my setting 
OBS:  Resolution: 640 x 480
Virtual-cam crop : All 4 is 0 ( because crop will effect output resolution)

And here is my Screenshot
[https://i.imgur.com/ZjP1n89.png](https://i.imgur.com/ZjP1n89.png)


I tried to use 640 x 360 but is not work , so maybe 16:9 resolution is not appropriate ?
I am not really sure now , but I will give an answer after some tests.

**Comment by MakarkinPRO** [comment]
Hmmm still the green screen - https://www.screencast.com/t/FQUIACxxb6c - this WebCam0
this WebCam1 - https://www.screencast.com/t/E05xRfwSBLn

**Comment by MakarkinPRO** [comment]
but on preview is a black screen https://www.screencast.com/t/DBo2q9ZEb6d

**Comment by CatxFish** [comment]
Wow, it's weird.  Perhaps it need to reopen emulator or something to get it work.
I will tried to make some test and improvement this weekend.
Maybe you won't face these frustrations if you check this project next week ?

**Comment by MakarkinPRO** [comment]
ok just happy to help) I have Win 10 64 bit .

**Comment by CatxFish** [comment]
FYI, The new version should work without these annoying setting ,but the modification brings some new bugs. I will release it after I fix these bugs. It might in 2-3 days.

**Comment by MakarkinPRO** [comment]
ok, if you need any tests let me know.

**Comment by CatxFish** [comment]
version 1.1.1 should work with Genymotion , but the default camera app in Genymotion force camera run on 640x480@ 10 or 15 fps  in my environment. I am not sure if it's sufficient to do what you want or maybe other apps will get better quality?

**Comment by MakarkinPRO** [comment]
Yeah it looks starting to work 
https://www.screencast.com/t/fqAvuXF2iZrh
probably it's okay. Yeah my main purpose is to Run it with Instagram, right now have issue of logging, probably that's I'm running at x86 not ARM machine.

**Comment by MakarkinPRO** [comment]
Any thought about virtual machine?
Or I have another plan is it possible to connect Android phone via USB to Win10 PC and choose my VirtualOBS camera on the real android device for streaming?

**Comment by CatxFish** [comment]
I don't clearly know what is your issue now. 
And virtual-cam is a software interface on Windows not a usb device , any scheme about using usb controller would not compatible with this project.
I don't think this is a workable solution.

**Comment by MakarkinPRO** [comment]
just need translate to instgram my OBS content with a good quality)

BTW instagram running but not able to login me in)

**Comment by MakarkinPRO** [comment]
Any expiriments with VitrualBox + (may be instagram) streaming and OBS + you virtual camera? )

**Comment by CatxFish** [comment]
I don't really use an emulator to streaming , I just test for the emulator by default camera app.
Is the problem occurring on camera ? I just thought you were telling that it's a android app issue cause you can't login successfully.

**Comment by Shubham0Rajput** [comment]
is it possible to add OBS Virtual Camera in VirtualBox, I want to send a custom camera video feed to Virtualbox.
